### PR TITLE
Allows skateboards to be picked up and used as weapons, new sprites

### PR DESCRIPTION
**Pull request by PKPenguin321**
Sprites by JStheguy
ss13shitboard

Pick up a skateboard by dragging it to yourself. Use a lifted skateboard inhand to place it back on the ground.

Boards are about as big as spears, do 12 damage, and are one-handed.

:cl: PKPenguin321
rscadd: You can now pick up a skateboard and use it as a weapon by dragging it to yourself.
imageadd: Skateboards have been given brand new sprites, courtesy of JStheguy
/:cl:

Merge Token: https://github.com/tgstation/tgstation/pull/18154

### CODE REVIEW

File: code/game/objects/items/weapons/weaponry.dm [hunk 1]
```diff
@@ -286,3 +286,17 @@
...
+	icon_state = "skateboard"
+	item_state = "skateboard"
+	force = 12
+	throwforce = 4
+	w_class = 5.0
+	attack_verb = list("smacked", "whacked", "slammed", "smashed")
+
+/obj/item/weapon/melee/skateboard/attack_self(mob/user)
+	new /obj/vehicle/scooter/skateboard(get_turf(user))
+	qdel(src)
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/weaponry.dm]: Why not have the object stored inside the vehicle instead of constantly deleting both?

Comment (PKPenguin321) [inline code/game/objects/items/weapons/weaponry.dm]: what advantages would that give? it feels like that would just cause problems

Comment (KorPhaeron) [inline code/game/objects/items/weapons/weaponry.dm]: Not constantly spawning and deleting uneeded objects is the advantage

Comment (Dorsisdwarf) [inline code/game/objects/items/weapons/weaponry.dm]: Also allows varediting the board to carry over

**Comment by pudl** [comment]
make them embed

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (PKPenguin321) [inline code/game/objects/items/weapons/weaponry.dm]: i can't think of a clean way of doing this. the best i've got is making a new path of skateboards that creates a board weapon inside it on New(), but that requires repathing all existing skateboards and map edits give me cancer

**Comment by KorPhaeron** [comment]
do maps actually ahve saketboards on them

**Comment by PKPenguin321** [comment]
i know at least 1 lavaland ruin has one

**Comment by KorPhaeron** [comment]
lavaland is trahs just ignore it

**Comment by KorPhaeron** [comment]
wait this doesn't require repathing though

**Comment by PKPenguin321** [comment]
??? well i don't know how to do that but what i already have here works

**Comment by KorPhaeron** [comment]
/obj/vehicle/scooter/skateboard/New()
..()
object_version = new obj/item/melee/skateboard(src)

And then mousedrop puts object_version in your hand and puts the board inside the object, and vice versa

File: code/game/objects/items/weapons/weaponry.dm (hunk 1 above)
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: NO CHANGE NO MERGE

Comment (PKPenguin321) [inline code/game/objects/items/weapons/weaponry.dm]: because i'm retarded and overcomplicated it in my head

**Comment by Steelpoint** [comment]
mfw a skateboard deals more damage than a stun baton or a toolbox

**Comment by Militaires** [comment]
@Steelpoint steelpoint stun batons are metal rods with electricity on them, they should give equal or maybe even less brute damage than a metal rod, and no burn damage unless you emag it.

File: code/modules/vehicles/scooter.dm [hunk 2]
```diff
@@ -74,6 +74,18 @@
 		visible_message("<span class='danger'>[src] crashes into [A], sending [H] flying!</span>")
 		playsound(src, 'sound/effects/bang.ogg', 50, 1)
 
+/obj/vehicle/scooter/skateboard/MouseDrop(atom/over_object)
+	var/mob/living/carbon/M = usr
+	if(!istype(M) || M.incapacitated() || !Adjacent(M))
+		return
+	if(has_buckled_mobs() && over_object == M)
+		M << "<span class='warning'>You can't lift this up when somebody's on it.</span>"
```
Comment (RandomMarine) [inline code/modules/vehicles/scooter.dm]: >No check for hulk to allow picking it up while flinging the poor sap away like he beefed it
It's like you hate fun.

**Comment by optimumtact** [comment]
come on finish

**Comment by PKPenguin321** [comment]
reee it's only two days old not even stale you dip

**Comment by PKPenguin321** [comment]
i tried what you asked and now there is an infinite recursion error send help
http://pastebin.com/r5BhtkTy

File: code/modules/vehicles/scooter.dm [hunk 3]
```diff
@@ -82,9 +88,8 @@
 		M << "<span class='warning'>You can't lift this up when somebody's on it.</span>"
 		return
 	if(over_object == M)
-		var/obj/item/weapon/melee/skateboard/board = new /obj/item/weapon/melee/skateboard()
-		M.put_in_hands(board)
-		qdel(src)
+		loc = linked_board
```
Comment (KorPhaeron) [inline code/modules/vehicles/scooter.dm]: forcemove

File: code/game/objects/items/weapons/weaponry.dm [hunk 4]
```diff
@@ -296,7 +296,15 @@
...
+	var/obj/vehicle/scooter/skateboard/linked_board
 
-/obj/item/weapon/melee/skateboard/attack_self(mob/user)
-	new /obj/vehicle/scooter/skateboard(get_turf(user))
-	qdel(src)
+/obj/item/weapon/melee/skateboard/attack_self(mob/user)//get_turf(user)
+	if(!user.drop_item())
+		return
+	linked_board.loc = get_turf(user)
+	loc = linked_board
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/weaponry.dm]: forcemove

**Comment by KorPhaeron** [comment]
Should probably have them delete eachother in Destroy as well

**Comment by optimumtact** [comment]
Each item has a new that spawns the other item, calilng it's new, spawning the other item, calling it's new, spawning the other item,  calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item, calling it's new, spawning the other item,

Stack overflow

File: code/game/objects/items/weapons/weaponry.dm [hunk 5]
```diff
@@ -286,3 +286,25 @@
...
+
+/obj/item/weapon/melee/skateboard/attack_self(mob/user)//get_turf(user)
+	if(!user.drop_item())
+		return
+	linked_board.loc = get_turf(user)
+	loc = linked_board
+
+/obj/item/weapon/melee/skateboard/New()
+	..()
+	linked_board = new /obj/vehicle/scooter/skateboard(src)
```
Comment (KorPhaeron) [inline code/game/objects/items/weapons/weaponry.dm]: you don't need this here

File: code/game/objects/items/weapons/weaponry.dm [hunk 6]
```diff
@@ -286,3 +286,25 @@
...
+	force = 12
+	throwforce = 4
+	w_class = 5.0
+	attack_verb = list("smacked", "whacked", "slammed", "smashed")
+	var/obj/vehicle/scooter/skateboard/linked_board
+
+/obj/item/weapon/melee/skateboard/attack_self(mob/user)//get_turf(user)
+	if(!user.drop_item())
+		return
+	linked_board.loc = get_turf(user)
```
Comment (optimumtact) [inline code/game/objects/items/weapons/weaponry.dm]: create the linked board here if not already set

File: code/modules/vehicles/scooter.dm [hunk 7]
```diff
@@ -74,6 +80,17 @@
...
+/obj/vehicle/scooter/skateboard/MouseDrop(atom/over_object)
+	var/mob/living/carbon/M = usr
+	if(!istype(M) || M.incapacitated() || !Adjacent(M))
+		return
+	if(has_buckled_mobs() && over_object == M)
+		M << "<span class='warning'>You can't lift this up when somebody's on it.</span>"
+		return
+	if(over_object == M)
+		loc = linked_board
+		M.put_in_hands(linked_board)
```
Comment (optimumtact) [inline code/modules/vehicles/scooter.dm]: create the item board here if not already set

**Comment by PKPenguin321** [comment]
well i fixed the recursion thing but now this really weird bug is happening and i can't figure out why

if you drag the board to yourself it puts it in your hand but the vehicle remains on the ground. if you drag it to yourself again, then it moves the board to your other hand (it looks like there are two boards, so it's really desyncing) and then properly removes the board vehicle. [i made a webm in case i wasn't clear](http://puu.sh/pvylA/34268477ea.webm)

**Comment by optimumtact** [comment]
hmm interesting

File: code/modules/vehicles/scooter.dm [hunk 8]
```diff
@@ -74,6 +75,25 @@
...
+	if(!istype(M) || M.incapacitated() || !Adjacent(M))
+		return
+	if(has_buckled_mobs() && over_object == M)
+		M << "<span class='warning'>You can't lift this up when somebody's on it.</span>"
+		return
+	if(over_object == M)
+		if(!linked_board)
+			linked_board = new /obj/item/weapon/melee/skateboard(src)
+			linked_board.linked_board = src
+		forceMove(linked_board)
```
Comment (optimumtact) [inline code/modules/vehicles/scooter.dm]: maybe this is failing the first time you try to mouse drop?, add some debugging output to see what exactly is going on.

Comment (PKPenguin321) [inline code/modules/vehicles/scooter.dm]: remind me why it's bad to delete and recreate objects and have a stable PR again

Comment (optimumtact) [inline code/modules/vehicles/scooter.dm]: because we told you so

**Comment by PKPenguin321** [comment]
i dont know what to do reeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeeee

**Comment by optimumtact** [comment]
I will try to look at this tonight

**Comment by optimumtact** [comment]
Ah you know what, I don't have time to debug the bug, so I think you should go back to what was working for you before and someone can clean it up later.

I don't think it's fair to deny your pr on that basis.

**Comment by PKPenguin321** [comment]
that should do it

**Comment by PKPenguin321** [comment]
it would be nice if this was merged now since if it has conflicts later i wont be able to fix them for the rest of the week

**Pull request merged**